clib/cMap.ml

```diff
@@ -65,6 +65,8 @@ module type ExtS = sig
 
   val fold_left : (key -> 'a -> 'b -> 'b) -> 'a t -> 'b -> 'b
   val fold_right : (key -> 'a -> 'b -> 'b) -> 'a t -> 'b -> 'b
+  val fold_left_map : (key -> 'a -> 'b -> 'c * 'b) -> 'a t -> 'b -> 'c t * 'b
```

Comment:
Nitpicking but since this is called `fold_left_map` and not `map_fold_left`, maybe
```suggestion
  val fold_left_map : (key -> 'a -> 'b -> 'b * 'c) -> 'a t -> 'b -> 'b * 'c t
```

Reply:
I chose `'a t -> 'b -> 'c t * 'b` thinking that it was morally a function from `'a t * 'b` to `'c t * 'b`.

We are already departing from the `List.fold_left_map` pattern, so I don't know any more on which intuitions it would be best to rely...

Reply:
Indeed, your argument makes sense, I don't know what's best.

Reply:
I personally prefer to have the accumulator on the left, it makes clearer what changes across the calls when the right-hand side is a complex expression. That said it's not a strong opinion either.

Reply:
Do you mean like Pierre or `(key -> 'b -> 'a -> 'b * 'c) -> 'b -> 'a t -> 'b * 'c t`? In the first case, it is 2-1 and I'll rally.

Reply:
Like @proux01